File: Modul-06/SOAL 4/PRAK604-2310817210009-MuhammadIbnuSina.py
def get_input():
    return input()

def count_characters(s):
    return len(s)

def print_comparison(pesan1, pesan2, count):
    bintang = 0
    pagar = 0

    for y in range(count):
        if pesan1[y] == ' ' and pesan2[y] == ' ':
            print(" ", end='')
        elif pesan1[y] == pesan2[y]:
            print("*", end='')
            bintang += 1
        elif pesan1[y] != pesan2[y]:
            print("#", end='')
            pagar += 1

    print("\n* =", bintang)
    print("# =", pagar)

    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

def main():
    array1 = get_input()
    array2 = get_input()

    count1 = len(array1)
    count2 = len(array2)

    if count1 == count2:
        pesan1 = array1
        pesan2 = array2

        print_comparison(pesan1, pesan2, count1)
    else:
        print("Panjang kalimat berbeda, pesan palsu")

if __name__ == "__main__":
    main()

File: Modul-06/SOAL 4/PRAK604-2310817210009-MuhammadIbnuSina.c
#include <stdio.h>

void getInput(char array[100]) {
    scanf("%[^\n]%*c", array);
}

int countCharacters(char array[100]) {
    int count = 0;
    for (int i = 0; i < 100; i++) {
        if (array[i] == '\0') {
            break;
        } else {
            count++;
        }
    }
    return count;
}

void printComparison(char pesan1[], char pesan2[], int count) {
    int bintang = 0;
    int pagar = 0;
    for (int y = 0; y < count; y++) {
        if (pesan1[y] == ' ' && pesan2[y] == ' ') {
            printf(" ");
        } else if (pesan1[y] == pesan2[y]) {
            printf("*");
            bintang++;
        } else if (pesan1[y] != pesan2[y]) {
            printf("#");
            pagar++;
        }
    }
    printf("\n");
    printf("* = %d\n", bintang);
    printf("# = %d\n", pagar);
    if (bintang >= pagar) {
        printf("Pesan Asli");
    } else {
        printf("Pesan Palsu");
    }
}

int main() {
    char array1[100];
    char array2[100];

    getInput(array1);
    getInput(array2);

    int count1 = countCharacters(array1);
    int count2 = countCharacters(array2);

    if (count1 == count2) {
        char pesan1[count1];
        char pesan2[count2];

        for (int x = 0; x < count1; x++) {
            pesan1[x] = array1[x];
            pesan2[x] = array2[x];
        }

        printComparison(pesan1, pesan2, count1);
    } else {
        printf("Panjang kalimat berbeda, pesan palsu");
    }

    return 0;
}
